.github/workflows/analyze.yml

```diff
@@ -0,0 +1,33 @@
+name: Analyze
+on:
+  pull_request:
+  push:
+    branches:
+      - confluence
+      - next
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      security-events: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          # Needed for commands that depend on git tags
+          fetch-depth: 0
```

Comment:
there were some issues w/ checkout action that were faced in go-livepeer. i'd recommend:
```yaml

      - name: Check out code
        uses: actions/checkout@v3
        with:
          fetch-depth: 0
          # Check https://github.com/livepeer/go-livepeer/pull/1891
          # for ref value discussion
          ref: ${{ github.event.pull_request.head.sha }}
```
like in https://github.com/livepeer/go-livepeer/blob/master/.github/workflows/build.yaml